File: java/src/com/jhobot/commands/info/GuildInfo.java
package com.jhobot.commands.info;

import com.jhobot.JhoBot;
import com.jhobot.handle.Messages;
import com.jhobot.handle.DB;
import com.jhobot.handle.Util;
import com.jhobot.handle.commands.Command;
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageReceivedEvent;
import sx.blah.discord.handle.obj.IGuild;
import sx.blah.discord.handle.obj.IUser;
import sx.blah.discord.handle.obj.Permissions;
import sx.blah.discord.util.EmbedBuilder;

import java.awt.*;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.Random;

public class GuildInfo implements Command {
    @Override
    public Runnable run(MessageReceivedEvent e, List<String> args) {
        return () -> {
            IGuild g = e.getGuild();
            EmbedBuilder b = new EmbedBuilder();
            b.withTitle("Guild : " + g.getName());
            b.appendField("Owner", g.getOwner().getName(), true);
            b.appendField("Role Amount", Integer.toString(g.getRoles().size()), true);
            int human = 0;
            int bot = 0;
            for (IUser u : g.getUsers())
            {
                if (u.isBot())
                    human++;
                else
                    bot++;
            }
            b.appendField("Humans", Integer.toString(human), true);
            b.appendField("Bots", Integer.toString(bot), true);
            b.appendField("User Amount", Integer.toString(g.getUsers().size()), true);
            b.appendField("Voice Channels", Integer.toString(g.getVoiceChannels().size()), true);
            b.appendField("Text Channels", Integer.toString(g.getChannels().size()), true);
            b.appendField("Categories", Integer.toString(g.getCategories().size()), true);
            Date date = Date.from(e.getGuild().getCreationDate());
            SimpleDateFormat format = new SimpleDateFormat("MM/dd/yyyy");
            b.appendField("Creation Date", format.format(date), false);

            b.withImage(e.getGuild().getIconURL());
            b.withColor(new Color(new Random().nextFloat(), new Random().nextFloat(), new Random().nextFloat()));
            b.withFooterText(Util.getTimeStamp());
            new Messages(e.getChannel()).sendEmbed(b.build());
        };
    }

    @Override
    public Runnable help(MessageReceivedEvent e, List<String> args) {
        return () -> {
            EmbedBuilder b = new EmbedBuilder();
            b.withTitle("Help : GuildInfo");
            b.appendField(JhoBot.db.getString(e.getGuild(), "prefix") + "guildinfo", "Gives information about the current guild.", false);
            b.withFooterText(Util.getTimeStamp());
            b.withColor(new Color(new Random().nextFloat(), new Random().nextFloat(), new Random().nextFloat()));
            new Messages(e.getChannel()).sendEmbed(b.build());
        };
    }
}

File: chad/src/main/java/dev/shog/chad/commands/info/GuildInfo.kt
package dev.shog.chad.commands.info

import dev.shog.chad.framework.handle.GuildHandler
import dev.shog.chad.framework.handle.MessageHandler
import dev.shog.chad.framework.obj.Command
import dev.shog.chad.framework.obj.Guild
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent
import sx.blah.discord.util.EmbedBuilder
import java.text.SimpleDateFormat
import java.util.*

/**
 * Gets info about the current guild
 *
 * @author sho
 */
class GuildInfo : Command.Class {
    override suspend fun run(e: MessageEvent, args: MutableList<String>) {
        // Creates an embed builder and applies the title
        val embedBuilder = EmbedBuilder()

        // Gets the human to bot.
        var human = 0
        var bot = 0
        for (u in e.guild.users) {
            if (!u.isBot)
                human++
            else
                bot++
        }

        val guild = GuildHandler.getGuild(e.guild.longID)

        // Applies the description
        embedBuilder.withDesc(
                "**Name** : `${e.guild.name}`" +
                        "\n**Owner** : `${e.guild.owner.name}`" +
                        "\n**Role Amount** : `${e.guild.roles.size}`" +
                        "\n**Human to Bots** : `$human/$bot`"+
                        "\n**User Amount** : `${e.guild.users.size}`" +
                        "\n**Voice Channels** : `${e.guild.voiceChannels.size}`" +
                        "\n**Text Channels** : `${e.guild.channels.size}`" +
                        "\n**Categories** : `${e.guild.categories.size}`" +
                        "\n**Creation Date** : `${SimpleDateFormat("MM/dd/yyyy").format(Date.from(e.guild.creationDate))}`" +
                        "\n\n**Messages Sent** : `${guild.getObject(Guild.DataType.MESSAGES_SENT)}`" +
                        "\n\n**Commands Sent** : `${guild.getObject(Guild.DataType.COMMANDS_SENT)}`"
        )

        // Adds the guild's image and sends.
        embedBuilder.withImage(e.guild.iconURL)
        MessageHandler(e.channel, e.author).sendEmbed(embedBuilder)
    }

    override suspend fun help(e: MessageEvent) {
        val st = HashMap<String, String>()
        st["guildinfo"] = "Gets information about the guild."
        Command.helpCommand(st, "Guild Info", e)
    }
}
